module/config.mjs

```diff
@@ -1078,6 +1078,31 @@ DND5E.SPELL_SLOT_TABLE = [
 
 /* -------------------------------------------- */
 
+/**
+ * Configuration data for pact casting progression.
+ *
+ * @typedef {object} PactProgressionConfig
+ * @property {number} slots  Number of spell slots granted.
+ * @property {number} level  Level of spells that can be cast.
+ */
+
+/**
+ * Define the pact slot & level progression by pact caster level.
+ * @enum {PactProgressionConfig}
+ */
+DND5E.pactCastingProgression = {
```

Comment:
Part of me is wondering if this is a perfect opportunity to make a nested object to allow for multiple pact progressions.
```js
DND5E.pactCastingProgression = {
  regular: {...},
  someOther: {...}
};
```

Reply:
It seems to me that would be better handled by custom spellcasting types using the hooks.